keymanagementscenarios.md

```diff
@@ -0,0 +1,139 @@
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin delegates access for build machine (Publisher Builder and Signer in this context) to create new delegate keys to be used on build machine. The delegate key can be stored on the build machine, USB Token, Network HSM, or Cloud Key Management Service (similar justification to earlier scenarios).
+    - Publisher Admin configures build scripts with credentials and if needed location of delegate keys.
+    - Build machine runs docker build and then notary v2 sign with keys from configured location.
+- Signing by Dedicated Machines
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin delegates access for signing machine (Publisher Signer in this context) to create new delegate keys to be used on build machine. The delegate key can be stored on the build machine, USB Token, Network HSM, or Cloud Key Management Service (similar justification to earlier scenarios).
+    - Publisher Admin configures signing machine with credentials and if needed location of delegate keys.
+    - Build machine (Publisher Builder only in this context) runs docker build and passes objects to be signed to signing host (the build machine does not have access to signing keys. This prevents signing keys from being exposed through potential vulnerabilities in build logic).
+    - Signing host generates signatures with keys in configured location.
+- Signing in Air Gapped Environments with Shared Root 
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin creates new subordinate key in air gapped environment on a host, USB Token, Network HSM, or Cloud Key Management Service chaining to root (one time operation enables the root to be used by runtime environments outside of the air gapped environment to validate signatures generated inside the environment).
+    - Publisher Admin configures credentials for Publisher Signers in air gapped environment to use sub-ordinate key for delegate keys instead of root key (users in air gapped environment do not need to get keys from outside of the environment). 
+
+Deployment Use Cases:
+
+Provisioning Truststore for Deployment
+
+![Notaryv2 Deployment Provisioning](./media/nv2-keymgmt-architecture-provisiontruststore-v1.png)
+
+Deployment Workflow
+
+![Notaryv2 Deployment Workflow](./media/nv2-keymgmt-architecture-pull-v1.png)
+
+Detailed Usecases:
+- Specify trusted public root keys
+    - Deployer Admin gets root public key from publisher.
+    - Deployer Admin adds root public key to runtime configuration.
+    - Container pulled from any registry is be validated with listed root public keys before execution.
```

Comment:
How does the user know to trust the listed root public keys? Can they chain trust from pre-configured keys?